controller/RestEligibilities.php

```diff
@@ -0,0 +1,144 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+namespace oat\taoTestCenter\controller;
+
+use common_exception_MissingParameter;
+use common_exception_NotFound;
+use common_exception_RestApi;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoTestCenter\model\eligibility\Eligibility;
+use oat\taoTestCenter\model\EligibilityService;
+
+class RestEligibilities extends AbstractRestController
+{
+
+    const PARAMETER_DELIVERY_ID = 'delivery';
+
+    /**
+     * @throws \common_exception_NotImplemented
+     */
+    public function post()
+    {
+        $this->returnFailure(new common_exception_RestApi('Not implemented.'));
+    }
+
+    /**
+     * @OA\Get(
+     *     path="/taoTestCenter/api/eligibilities",
+     *     tags={"eligibilities"},
+     *     summary="Search for eligibilities",
+     *     description="Search for eligibilities",
+     *     @OA\Parameter(
+     *         name="delivery",
+     *         in="query",
+     *         description="Delivery Uri (Url encoded)",
+     *         required=true,
+     *         @OA\Schema(
+     *             type="string",
+     *         )
+     *     ),
+     *     @OA\Parameter(
+     *         name="testCenter",
+     *         in="query",
+     *         description="Test center Uri (Url encoded)",
+     *         required=true,
+     *         @OA\Schema(
+     *             type="string",
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response="200",
+     *         description="Eligibility data",
+     *         @OA\MediaType(
+     *             mediaType="application/json",
+     *             @OA\Schema(
+     *                 ref="#/components/schemas/Eligibility",
+     *                 example= {
+     *                     "delivery": "http://sample/first.rdf#i1536680377163170",
+     *                     "testCenter": "http://sample/first.rdf#i1536680377163171",
+     *                     "testTakers": {
+     *                         "http://sample/first.rdf#i1536680377163172",
+     *                         "http://sample/first.rdf#i1536680377163173"
+     *                     }
+     *                 }
+     *             )
+     *         ),
+     *     ),
+     *     @OA\Response(
+     *         response=404,
```

Comment:
It will never return 404 as the empty response is handled in `RestEligibilites::get()` in line 111.